src/System.Private.CoreLib/src/System/Threading/Overlapped.cs

```diff
@@ -287,6 +288,9 @@ public static unsafe void Free(NativeOverlapped* nativeOverlappedPtr)
             {
                 if (!success)
                     FreeNativeOverlapped();
+
+                if (success && FrameworkEventSource.Log.IsEnabled(EventLevel.Verbose, FrameworkEventSource.Keywords.ThreadPool))
+                    System.Diagnostics.Tracing.FrameworkEventSource.Log.ThreadPoolIOPackWork((long)(IntPtr)_pNativeOverlapped);
```

Comment:
This needs to be checked to see if it produces the same ID as the native code to allow the events to be properly correlated.

Reply:
It doesn't contain all the relevant data yet. I finally got to check it... Working on it.